xml/tuning_cgroups.xml

```diff
@@ -286,246 +286,225 @@ TasksMax=16284
-    <option>cgroup_no_v1=blkio</option>. Verify that this parameter was used,
-    and that the IO controller (cgroup-v2) is available:
-   </para>
-<screen>
-&prompt.io-controller;cat /proc/cmdline
-BOOT_IMAGE=... cgroup_no_v1=blkio ...
-&prompt.io-controller;cat /sys/fs/cgroup/unified/cgroup.controllers
-io
-</screen>
-   <para>
-    Next, enable the IO controller:
-   </para>
-<screen>
-&prompt.io-controller;cd /sys/fs/cgroup/unified/
-&prompt.unified;echo '+io' > cgroup.subtree_control
-&prompt.unified;cat cgroup.subtree_control
-io
-</screen>
-   <para>
-    Now run all the test steps, similarly to the steps for cgroup-v1. Note that
-    some of the directories are different. Run a single reader process (in the
-    examples, reading from an SSD) without controlling its I/O, using
-    file2.txt:
-   </para>
+<sect3 xml:id="sec-cgroups-unified-hierarchy">
+<title>Unified cgroup hiearchy</title>
+<para>
+To properly account also writeback I/O, it is necessary to have equal I/O
+and memory controller cgroup hierarchies and use cgroup v2 I/O controller.
+Together, this means that one has to use the <emphasis>unified</emphasis>
+cgroup hierarchy. This has to be requested explicitly in SLES by passing a
+kernel command line option <option>systemd.unified_cgroup_hierarchy=1</option>.
+</para>
+</sect3>
+
+<sect3 xml:id="sec-cgroups-block-io-scheduler">
+<title>Block I/O scheduler</title>
+<para>
+The throttling policy is implemented higher in the stack therefore it
+doesn’t require any additional adjustments.
+The proportional I/O control policies have two different implementations:
+BFQ controller and cost-based model.
+We describe the BFQ controller here and in order to exert its proportional
+implementation for a particular device, we must make sure that BFQ is the
+chosen scheduler.
+Check the current scheduler:
+</para>
 <screen>
-&prompt.unified;cd -
-&prompt.io-controller;sync; echo 3 > /proc/sys/vm/drop_caches
-&prompt.io-controller;echo $$; dd if=file2.txt of=/dev/null bs=4k count=131072
-5633
-[...]
+&prompt.user;<command>cat /sys/class/block/$DEV/queue/scheduler</command>
+mq-deadline kyber bfq [none]
 </screen>
-   <para>
-    Run a background process reading from the same disk and note your
-    throughput values:
-   </para>
+<para>
+Switch the scheduler:
+</para>
 <screen>
-&prompt.reader1;sync; echo 3 > /proc/sys/vm/drop_caches
-&prompt.reader1;echo $$; dd if=file1.txt of=/dev/null bs=4k
-5633
-[...]
-&prompt.reader2;echo $$; dd if=file2.txt of=/dev/null bs=4k count=131072
-5703
-[...]
+ &prompt.root;<command>echo bfq >/sys/class/block/<replaceable>sda1</replaceable>/queue/scheduler</command>
 </screen>
-   <para>
-    Each process gets half of the throughput for I/O operations. Set up two
-    control groups&mdash;one for each process&mdash;verify that BFQ is the
-    active scheduler, and set a different weight for reader2:
-   </para>
+<para>
+<literal>$DEV</literal> is the disk device, not a partition and the
```

Comment:
This should be the same replaceable as in the code block above.
But more importantly, it should be only `sda` (without the partition number).